src/components/ServiceRequestForm.jsx

```diff
@@ -154,33 +174,21 @@ const ServiceRequestForm = (props, errors, touched) => {
 		}
 
 		pullServiceRequestFields(fields);
-	};
-
-	const pullServiceRequestFields = (fields) => {
-		if (fields !== undefined) {
-			localProps.setFieldValue('requestTypeAddressID', fields.streetAddress);
-			localProps.setFieldValue('requestTypeCityID', fields.city);
-			localProps.setFieldValue('requestTypeZipID', fields.zipCode);
-			localProps.setFieldValue('isPanelRequired', true);
-		}
-		else{
-			localProps.setFieldValue('isPanelRequired', false);
-		}
-	};
-
-	const handleServiceSubRequestChange = (changeEvent) => {
+	}
 
-		const value = changeEvent.currentTarget.value.toLowerCase();
-		const subInfo = getSubCategoriesIncludedDescription(subCategories, value ? value : value);
+	const addSelectedSubValueOptions = (Categories, value)=>{
+		const subCategories = Categories.flatMap(x => x.types);
+		const subInfo = getSubCategoriesIncludedDescription(subCategories, value);
```

Comment:
not for this pr, but there is now  CategoryHelper that provides this functionality in one  call `GetSubCategory`.

```js
var subCategory = GetSubCategory(categories, 1234356);

// subCategory will be
/*
{
	"name": "Other website problem",
	"id": 1234356,
	"note": "In the description below, please include the URL of the page where you encountered the problem.",
        "shouldDisableForm": true
}
*/

this will simplify this logic a bit